apluslms_roman/backends/__init__.py

```diff
@@ -38,17 +39,21 @@ def from_config(cls, index, data):
                 raise RuntimeError("Missing image name (img) in step configuration: {}".format(data))
             return cls(
                 index,
-                clean_image_name(data['img']),
+                data.get('img'),
                 data.get('cmd'),
                 data.get('mnt'),
-                dict(data.get('env', {})),
+                data.get('env'),
                 data.get('name'),
             )
         else:
             return cls(index, clean_image_name(data))
 
     def __init__(self, ref, img, cmd=None, mnt=None, env=None, name=None):
-        self.img, self.cmd, self.mnt, self.env, self.name, self.ref = img, cmd, mnt, env, name, ref
+        self.img = clean_image_name(img)
+        self.env = dict(env) if env else {}
+        if isinstance(cmd, ChangesList):
```

Comment:
`if not isinstance(cmd, str)`. Try to avoid importing ChangesList to different locations. In addition, it might be better to cast cmd to `tuple`, so it isn't so easy to edit it by accident.